hkmc2/shared/src/main/scala/hkmc2/codegen/Lifter.scala

```diff
@@ -234,55 +236,38 @@ class Lifter(handlerPaths: Opt[HandlerPaths])(using State, Raise):
     val FreeVars(_, cap) = ctx.usedLocals(f.sym)
 
     val fresh = FreshInt()
-
-    val varsMap = cap.map: s =>
-        val id = fresh.make
-        val nme = s.nme + id + "$"
-        val varSym = VarSymbol(Tree.Ident(nme))
-        val fldSym = BlockMemberSymbol(nme, Nil)
-        val fldDef = TermDefinition(
-          S(clsSym),
-          syntax.ImmutVal,
-          fldSym,
-          Nil, N, N,
-          S(Term.Ref(s)(Tree.Ident(s.nme), 666)), // FIXME: 666 is a dummy value
-          FlowSymbol("‹class-param-res›"),
-          TermDefFlags.empty,
-          Nil
-        )
-        fldSym.defn = S(fldDef)
-        s -> (
-          varSym,
-          fldDef,
-        )
-      .toMap
-
-    val varsList = cap.toList
```

Comment:
You can't `toList` a set to get some arbitrary order, because that order is not well-defined. It could change depending on the object's memory address next time the compiler runs after minor modifications. But we need stable and reproducible compilation outputs.